power_grid_model_c/power_grid_model/include/power_grid_model/calculation_parameters.hpp

```diff
@@ -15,8 +15,8 @@ namespace power_grid_model {
 
 // Entry of YBus, node addmittance matrix
 struct YBusElement {
-    YBusElementType element_type;
-    Idx idx; // index of the component
+    YBusElementType element_type{};
+    Idx idx{}; // index of the component
```

Comment:
It this desired? If you make default value for a `struct`, it is not aggregate type anymore. So you cannot do

```c++
YBusElement element{.element_type = 1, .idx=5};
```

Reply:
shouldn't matter: https://godbolt.org/z/o814Kff7q

Reply:
in fact, if you don't have a default initializer, you will receive compilation errors when you do not initialize all fields. Try removing `.b = 2` when constructing `A` in https://godbolt.org/z/3Y7c65WEo

The only argument against default initializing `{}` is that it doesn't default initialize to `nan` (or equivalent per type) but we can decide on adding that now

Reply:
Okay good. Then leave it as is.

Reply:
FYI, only under C++11, the type will become non-aggregate if you have default initializer for data member. See
https://godbolt.org/z/87fsr18bx

This constraint has been removed since C++14.
https://en.cppreference.com/w/cpp/language/aggregate_initialization